omnigibson/systems/micro_particle_system.py

```diff
@@ -95,15 +100,17 @@ def __init__(self, prim_path, name, idn):
         self._idn = idn
 
         # Run super method directly
-        super().__init__(prim_path=prim_path, name=name)
-
-        self._parent_prim = BasePrim(prim_path=self.prim.GetParent().GetPath().pathString, name=f"{name}_parent")
+        super().__init__(relative_prim_path=relative_prim_path, name=name)
 
     def _load(self):
         # We raise an error, this should NOT be created from scratch
         raise NotImplementedError("PhysxPointInstancer should NOT be loaded via this class! Should be created before.")
 
     def remove(self):
+        parent_absolute_path = self.prim.GetParent().GetPath().pathString
+        parent_relative_path = absolute_prim_path_to_scene_relative(self._scene, parent_absolute_path)
+        self._parent_prim = BasePrim(relative_prim_path=parent_relative_path, name=f"{self._name}_parent")
+        self._parent_prim.load(self._scene)
```

Comment:
I don't think I quite understand this logic here. Why does a prim need to be created before removing this system (which is also then deleted immediately afterwards)?

Reply:
Here's what we used to have: 
1. we create the parent prim at init time (we can't do it now because of our new setup - we can only access the prim after load)
2. then, in remove, we get rid of the parent prim (this is the only place where this parent prim is used)

Now, essentially we are just delaying the parent prim creation until when we actually need it for removal.

Reply:
TODO: add a comment - avoid low level delete prim method